### PR TITLE
much prettier, still needs organization

### PR DESCRIPTION
**Pull request by Fritzzzz1**

### CODE REVIEW

File: games/helpers.py [hunk 1]
```diff
@@ -1,19 +1,37 @@
 from boards.serializer import BoardSerializer
 from attempts.serializer import AttemptSerializer
+from games.serializer import GameSerializer
+from boards.models import Board
+from PVP_WORDLE_BackEnd.helpers import check_and_save_serializer
```
Comment (TamirOr) [inline games/helpers.py:5]: I think PVP_WORDLE_BackEnd should be as package name - pvp_wordle_backend

File: games/helpers.py [hunk 2]
```diff
@@ -1,19 +1,37 @@
...
 
 
-def create_board(game_id):
-    board_serializer = BoardSerializer(data={'id': game_id, 'game': game_id})
-    if board_serializer.is_valid():
-        board_serializer.save()
+def handle_create_new_game(request_data, player_id):
```
Comment (TamirOr) [inline games/helpers.py:8]: Try to think of a better name for this function | create_game

File: games/helpers.py [hunk 3]
```diff
@@ -1,19 +1,37 @@
...
-        attempts.append({'id': 10 * game_id + i, 'board': game_id, 'word': words[i - 1]})
+    if not isinstance(board_pk_or_errors, int):
+        return board_pk_or_errors
 
-    attempt_serializer = AttemptSerializer(data=attempts, many=True)
-    if attempt_serializer.is_valid():
-        attempt_serializer.save()
+    words = request_data.get('attempts')
+    if words:
+        attempts = [{'board': board_pk_or_errors, 'word': words[i - 1]} for i in range(1, len(words) + 1)]
```
Comment (TamirOr) [inline games/helpers.py:22]: you dirty one ;)

File: pvp_wordle_backend/helpers.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+
+
+def check_and_save_serializer(data, serializer_class, context=None):
+
+    serializer = serializer_class(data=data, many=True if context is "many" else False)
+
+    if serializer.is_valid():
+        new_obj = serializer.save()
+        if isinstance(new_obj, list):
+            return 1
```
Comment (TamirOr) [inline pvp_wordle_backend/helpers.py:10]: change to ENUM(1 isn't informative enough)

File: boards/serializer.py [hunk 5]
```diff
@@ -6,4 +6,4 @@ class BoardSerializer(serializers.ModelSerializer):
 
     class Meta:
         model = Board
-        fields = ('id', 'game')
+        fields = '__all__'
```
Comment (TamirOr) [inline boards/serializer.py:9]: oh-hh so sophisticated

File: players/urls.py [hunk 6]
```diff
@@ -1,10 +1,13 @@
...
+app_name = 'players'
 urlpatterns = [
-    re_path(r'^api/players/$', players_list, name='players-list'),
-    re_path(r'^api/players/([0-9])$', players_detail, name='players-detail'),
-    re_path(r'^api/players/create/$', players_signup, name='players-signup'),
+    path('', player_list, name='player-list'),
+    path('<player_id>', player_detail, name='player-detail'),
+    path('create/', player_create, name='player-create'),
+    path('<int:player_id>/games/', player_game_list, name='player-game-list'),
+    # path('<int:player_id>/games/<int:game_id>', player_game_detail, name='player-game-detail'),
```
Comment (TamirOr) [inline players/urls.py:12]: Do not push comment-out lines

**Review by TamirOr — CHANGES_REQUESTED** [review]
codes look good.
check about .gitignore and add it,  I don't think you need to push __pycache__ files.

File: games/helpers.py (hunk 3 above)
Comment (Fritzzzz1) [reply games/helpers.py:22]: i dont get it

Comment (TamirOr) [reply games/helpers.py:22]: I like what you wrote here

Comment (Fritzzzz1) [reply games/helpers.py:22]: thank you

File: pvp_wordle_backend/helpers.py (hunk 4 above)
Comment (Fritzzzz1) [reply pvp_wordle_backend/helpers.py:10]: maybe True is good? or also not enough.

File: players/views.py [hunk 7]
```diff
@@ -1,53 +1,57 @@
...
-
-        for player in players:
-            data.append(player)
-            counter += 1
-            if counter >= 10:
-                break
-        '''
-        serializer = PlayerSerializer(players, context={'request': request}, many=True)
-        return Response(serializer.data)
+class GetToken(TokenObtainPairView):
```
Comment (TamirOr) [inline players/views.py:14]: Is it a view function?

File: players/views.py [hunk 8]
```diff
@@ -1,53 +1,57 @@
...
-        if serializer.is_valid():
-            serializer.save()
-            return Response(status=status.HTTP_201_CREATED)
-    return Response(status=status.HTTP_201_CREATED)
+@api_view(['GET'])
+def player_list(request):
+    players = Player.objects.all().exclude(username='guest').order_by('-score')[:10]
+    serializer = PlayerSerializer(players, many=True)
+    for player in serializer.data:
+        player['games'] = len(Player.objects.get(pk=player['pk']).game.all())
```
Comment (TamirOr) [inline players/views.py:23]: to me, map function is more readable then for loop.
for example:
    serializer.data.map(lambda player: player.update(games=len(Player.objects.get(pk=player['pk']).game.all())))

**Review by TamirOr — APPROVED** [review]

File: players/views.py (hunk 7 above)
Comment (Fritzzzz1) [reply players/views.py:14]: it is a class based view, used as a view function in players.urls with as_view()

File: players/views.py (hunk 8 above)
Comment (Fritzzzz1) [reply players/views.py:23]: you dirty dirty saying you don't really know python

Comment (TamirOr) [reply players/views.py:23]: :)

**Pull request merged**